### DISCUSSION TITLE
Can't join own local server

### DISCUSSION THREAD

**Discussion by bram2202** [mbround18/valheim-docker#618] [Help!] [answered]
After playing together with some friend run from one of ours game PC i wanted to run our own dedicated server so that we are not reliant on the one friend being online to continue our game.

I ran this image on my server, used the compose example and setup/startup was easy,
I setup port forwarding to my server and my friend can join the server without any problems using IP:PORT

The strange thing is that i can't, i tried my external IP (and port) that i handed out to my friend, and even the local IP of my server, and both times i wasn't able to connect.
All the version are the same.

I don't know if this a local network issue of other limitation.
 
 Did anyone ran into the same problem?

**Comment by bram2202** [comment, marked as answer]
AH, one thing that just popped into my mind.
I'm using Valheim downloaded via xbox game pass, and my friend is using the steam version.

**Reply by mbround18** [reply]
@bram2202 im unsure if they are cross playable but should be, on your local network you woul dhave to use the local IP. I join my local server with docker all the time. what sort of network do you have?

**Reply by bram2202** [reply]
Yesterday a second friend attempted to connect to the same server with the same IP:Port combo.
he uses an gamepass downloaded version of valheim, just like me.
And he to was unable to connect.
So it looks more like a cross issues then a network issue.

The 3 of us joined a game hosted by the one friend that uses the steam version, so you would think this would work too.

**Reply by mbround18** [reply]
@bram2202 got it! :) [Crossplay requires a flag](https://www.valheimgame.com/news/patch-0-211-7-crossplay/) i can investigate and add a feature for it

**Reply by mbround18** [reply]
@bram2202  Got an image up https://hub.docker.com/layers/mbround18/valheim/crossplay-test/images/sha256-299a0710b946c89a69e7805aa244c11f2e7d24d046473af42fa302eab150f188?context=explore

requires env variable set ENABLE_CROSSPLAY=1 testing it out atm

**Reply by mbround18** [reply]
I can get the server up with PlayFab which is the tool they use to connect but they dont spit out a join code 0.o 
```
10/09/2022 22:29:30: Server 'Valheim WooHoo PC Gaming' begin PlayFab create and join network for server 
```

**Reply by mbround18** [reply]
![image](https://user-images.githubusercontent.com/12646562/194803829-b2604374-8589-47d0-afca-e44631bdab5e.png)

**Reply by mbround18** [reply]
@bram2202 it was a missing dep, with the new multiplayer a new dep for the docker was needed, should work now if you wannna try the image `mbround18/valheim:crossplay-test` with `ENABLE_CROSSPLAY=1`. Ill submit a pr and git it underway for our process :)

**Reply by mbround18** [reply]
https://github.com/mbround18/valheim-docker/pull/619

**Reply by bram2202** [reply]
Nice! thank you very much.
I can test this with my friends in about 8 hours.

**Reply by bram2202** [reply]
@mbround18 we tried it, and it worked. Thanks!!

**Reply by mbround18** [reply]
@bram2202 1.7.0 is the version with that fix
